server/test/integration/com/thoughtworks/go/server/service/UserServiceIntegrationTest.java

```diff
@@ -609,6 +611,73 @@ public void shouldDisableAdminSelectionWhenUserIsMemberOfAdminRole() throws Exce
         assertThat(userService.getAdminAndRoleSelections(Arrays.asList("foo", "quux")).getAdminSelection(), is(new TriStateSelection(Admin.GO_SYSTEM_ADMIN, TriStateSelection.Action.add, false)));
     }
 
+    @Test
+    public void updateShouldUpdateEnabledState() throws Exception {
+        userDao.saveOrUpdate(new User("admin-user"));
+        userDao.saveOrUpdate(new User("user-1"));
+        userService.enable(Arrays.asList("user-1"), new HttpLocalizedOperationResult());
+
+        HttpLocalizedOperationResult result = new HttpLocalizedOperationResult();
+        userService.update(new User("user-1"), TriState.FALSE, TriState.UNSET, null, null, result);
+        assertThat(result.isSuccessful(), is(true));
+        assertThat(userService.findUserByName("user-1").isEnabled(), is(false));
+
+        result = new HttpLocalizedOperationResult();
+        userService.update(new User("user-1"), TriState.TRUE, TriState.UNSET, null, null, result);
+        assertThat(result.isSuccessful(), is(true));
+        assertThat(userService.findUserByName("user-1").isEnabled(), is(true));
+    }
+
+    @Test
+    public void updateShouldUpdateEmail() throws Exception {
+        userDao.saveOrUpdate(new User("user-1"));
+
+        HttpLocalizedOperationResult result = new HttpLocalizedOperationResult();
+        userService.update(new User("user-1"), TriState.UNSET, TriState.UNSET, "foo@example.com", null, result);
+        assertThat(result.isSuccessful(), is(true));
+
+        assertThat(userService.findUserByName("user-1").getEmail(), is("foo@example.com"));
+
+        result = new HttpLocalizedOperationResult();
+        userService.update(new User("user-1"), TriState.TRUE, TriState.UNSET, "", null, result);
+        assertThat(userService.findUserByName("user-1").getEmail(), is(""));
+        assertThat(result.isSuccessful(), is(true));
+    }
+    @Test
+    public void updateShouldUpdateMatcher() throws Exception {
+        userDao.saveOrUpdate(new User("user-1"));
+
+        HttpLocalizedOperationResult result = new HttpLocalizedOperationResult();
+        userService.update(new User("user-1"), TriState.UNSET, TriState.UNSET, null, "foo,bar", result);
+        assertThat(result.isSuccessful(), is(true));
+        assertThat(userService.findUserByName("user-1").getMatcher(), is("foo,bar"));
+
+        result = new HttpLocalizedOperationResult();
+        userService.update(new User("user-1"), TriState.UNSET, TriState.UNSET, null, "", result);
+        assertThat(result.isSuccessful(), is(true));
+        assertThat(userService.findUserByName("user-1").getMatcher(), is(""));
+    }
+
+    @Test
+    public void updateShouldUpdateEmailMeState() throws Exception {
+        userDao.saveOrUpdate(new User("user-1"));
+
+        User user = userService.findUserByName("user-1");
+        user.setEmailMe(false);
+        userDao.saveOrUpdate(user);
+
+        userService.enable(Arrays.asList("user-1"), new HttpLocalizedOperationResult());
```

Comment:
Should we enable the user for an update?

I think there are too many asserts in the test cases can't we split it up like this?

```
    @Test
    public void shouldUpdateEmailMeStateToTrue() throws Exception {
        User user = new User("user-1");
        user.setEmailMe(false);
        addUser(user);

        HttpLocalizedOperationResult result = new HttpLocalizedOperationResult();
        userService.update(user, TriState.UNSET, TriState.TRUE, null, null, result);

        assertThat(result.isSuccessful(), is(true));
        assertThat(userService.findUserByName("user-1").isEmailMe(), is(true));

    }
    @Test
    public void shouldUpdateEmailMeStateToFalse() throws Exception {
        User user = new User("user-1");
        user.setEmailMe(false);
        addUser(user);

        HttpLocalizedOperationResult result = new HttpLocalizedOperationResult();
        userService.update(user, TriState.UNSET, TriState.FALSE, null, null, result);

        assertThat(result.isSuccessful(), is(true));
        assertThat(userService.findUserByName("user-1").isEmailMe(), is(false));
    }
```

Also use the ```addUser``` method